### ISSUE TITLE
Error in Form Validation (cannot read property of null)

### ISSUE THREAD

**Issue by jetwes** [livewire/livewire#297]
**Describe the bug**
If a validation fails i get an error on echoing the actual errors

`livewire.js?id=1387c89dcf270acf8587:3411 Uncaught (in promise) TypeError: Cannot read property 'getAttributeNames' of null`

The validation error is shown, but the console shows this error and the livewire component is terminated. 

the code of my view:
```
<div class="mt-20 ml-10 mr-10">
    <h2 class="text-primary-600">Viermieter hinzufügen</h2>
    @if (count($errors) > 0)
        @foreach ($errors->all() as $error)
            <div class="alert bg-warning-400 alert-styled-left">
                {{ $error }}
            </div>
        @endforeach
    @endif
    <form wire:submit.prevent="createLandlord" enctype="multipart/form-data" class="form-vertical">
        @csrf
        <div class="form-group">
            <div class="row">
                <div class="col-md-3">
                    <label for="name" class="control-label">Name</label>
                    <input type="text" name="name" wire:model.lazy="name" id="name" class="form-control">
                </div>
                <div class="col-md-3">
                    <label for="street" class="control-label">Strasse</label>
                    <input type="text" name="street" id="street" class="form-control">
                </div>
                <div class="col-md-1">
                    <label for="zip" class="control-label">PLZ</label>
                    <input type="text" name="zip" id="zip" class="form-control">
                </div>
                <div class="col-md-2">
                    <label for="city" class="control-label">Ort</label>
                    <input type="text" name="city" id="city" class="form-control">
                </div>
                <div class="col-md-2">
                    <label for="contact_person" class="control-label">Kontaktperson</label>
                    <input type="text" name="contact_person" id="contact_person" class="form-control">
                </div>
            </div>
        </div>
        <div class="form-group">
            <div class="row">
                <div class="col-md-3">
                    <label for="phone" class="control-label">Telefon</label>
                    <input type="text" name="phone" id="phone" class="form-control">
                </div>
                <div class="col-md-3">
                    <label for="fax" class="control-label">Telefax</label>
                    <input type="text" name="fax" id="fax" class="form-control">
                </div>
                <div class="col-md-2">
                    <label for="email" class="control-label">E-Mail</label>
                    <input type="text" name="email" id="email" class="form-control">
                </div>
                <div class="col-md-2">
                    <label for="website" class="control-label">Website</label>
                    <input type="text" name="website" id="website" class="form-control">
                </div>
            </div>
        </div>
        <div class="form-group">
            <div class="">
                <button type="submit" id="newUserSubmit" class="btn bg-teal-400 "><i class="fa fa-check"></i> Speichern</button>
                <button type="reset" class="btn btn-danger" wire:click="setShowIndex(true)" aria-hidden="true">Abbrechen</button>
            </div>
        </div>
    </form>
</div>
```

The error does not occur if i remove the div:

```
@if (count($errors) > 0)
        @foreach ($errors->all() as $error)         
                {{ $error }}
        @endforeach
    @endif
```

**Comment by ServerJunge** [comment]
Same here...

Just used the example from the docs

https://laravel-livewire.com/docs/input-validation/
```
$validatedData = $this->validate([
            'email' => 'required|email',
        ]);  
```

```
 @if($errors->has('email'))
            <span class="mx-auto mt-2">{{ $errors->first('email') }}</span>
        @endif
```

**Comment by stefanbauer** [comment]
I can confirm that.

**Comment by stefanbauer** [comment]
I dig into it much more and I found at least the problem.

The problem on my side is/was I use the `@error` directive. It works fine when you use it only once. But if you have multiple fields and use that multiple times, it ends up with the error above. I understand, why this happens actually (because it tries to render `$message` several times in different contexts afaik), but I don't have any idea if that could be solved somehow.

The only and quick workaround for me now is not to use the `@error` directive and better go with the classic `@if` approach. Can this be solved by Livewire itself @calebporzio? Any ideas on that?

**Comment by stefanbauer** [comment]
Ok this was a little bit fast. That's not exactly the problem. Here is the problem:

This works:

```html
<input wire:model.lazy="email"/>
@if ($errors->has('email'))
    <div>{{ $errors->first('email') }}</div>
@endif

<input wire:model.lazy="password"/>
@if ($errors->has('password'))
    <div>{{ $errors->first('password') }}</div>
@endif
```

This doesn't work:

```html
<input wire:model.lazy="email"/>
@if ($errors->has('email'))
    <div>{{ $errors->first('email') }}</div>
@endif

<input class="@if ($errors->has('password')) bg-red @endif" wire:model.lazy="password"/>
@if ($errors->has('password'))
    <div>{{ $errors->first('password') }}</div>
@endif
```

As soon as you use a condition within an html tag it blows up.

**Comment by calebporzio** [comment]
Ok, I just tried to replicate the issue locally and couldn't.

Pull down `dev-master` and try it locally. It's possible a recent addition fixed this. There were issues similar to this recently resolved.

Thanks.

**Comment by stefanbauer** [comment]
It has been solved with the current `dev-master` (whatever you did) :)

**Comment by zajinx** [comment]
the error is back, it works ok the first time i hit AddUser, but if i want to add a new user as soon as i click in any of the input fields i get the error below, the error goes away if i remove the @if($this->success) block from the blade component.

```
 "php": "^7.2",
 "fideloper/proxy": "^4.0",
 "laravel/framework": "^6.2",
 "laravel/tinker": "^2.0",
 "livewire/livewire": "^0.5.2"
```

![image](https://user-images.githubusercontent.com/16177818/71383994-7e9b4700-25e7-11ea-9ea1-1e423a74b733.png)

```
<?php

namespace App\Http\Livewire\Users;

use Livewire\Component;
use App\User;
use Auth;
use Illuminate\Support\Facades\Hash;
use Illuminate\Support\Facades\Validator;
class Add extends Component
{

    public $name;
    public $email;
    public $password;
    public $password_confirmation;
    public $success;
    public $resultm;
    public $results;
    public $uniqid;

    public function mount(){
        $this->success=false;
        $this->uniqid=null;
        $this->results=null;
        $this->resultm=null;
    }
    public function updated($field)
    {
        $this->validateOnly($field, [
            'name' => ['required', 'string', 'max:255'],
            'email' => ['required', 'string', 'email', 'max:255', 'unique:users'],
            'password' => ['required', 'string', 'min:8', 'confirmed'],
        ]);
    }

    public function addUser()
    {
        $this->validate([
            'name' => ['required', 'string', 'max:255'],
            'email' => ['required', 'string', 'email', 'max:255', 'unique:users'],
            'password' => ['required', 'string', 'min:8', 'confirmed'],
        ]);

        $user=new User;
        $user->password=Hash::make($this->password);
        $user->name=$this->name;
        $user->email=$this->email;
        if($user->save()){
            $this->resultm="User Added";
            $this->results="success";
            $this->success=true;

            $this->name=null;
            $this->email=null;
            $this->password=null;
            $this->password_confirmation=null;
        }
        else{
            $this->resultm="Failed to add User";
            $this->results="danger";
            $this->success=false;
        }

    }

    public function render()
    {
        return view('livewire.users.add');
    }
}

```


```
<div class="card">
    <div class="card-header">User Add</div>
    @if($this->success)
    <div class="alert alert-primary alert-dismissible fade show" role="alert">
        <button type="button" class="close" data-dismiss="alert" aria-label="Close">
            <span aria-hidden="true">&times;</span>
            <span class="sr-only">Close</span>
        </button>
        <strong>Holy guacamole!</strong> You should check in on some of those fields below.
    </div>
    @endif

    <div class="card-body">
        <form wire:submit.prevent="addUser">
            <div class="form-group">
              <label for="{{$this->uniqid}}_name">Name</label>
              <input type="text"
                class="form-control @error('name') is-invalid @enderror" name="name" id="{{$this->uniqid}}name" aria-describedby="helpId" placeholder=""  wire:model="name">
              <small id="{{$this->uniqid}}_helpId" class="form-text text-muted">your name</small>
              @error('name') <div  class="invalid-feedback">{{ $message }}</div> @enderror
            </div>
            <div class="form-group">
              <label for="{{$this->uniqid}}_email">Email</label>
              <input type="text"
                class="form-control @error('email') is-invalid @enderror" name="email" id="{{$this->uniqid}}email" aria-describedby="helpId" placeholder="" wire:model="email">
              <small id="{{$this->uniqid}}_helpId" class="form-text text-muted">your email address</small>
              @error('email') <div  class="invalid-feedback">{{ $message }}</div> @enderror
            </div>
            <div class="form-group">
              <label for="{{$this->uniqid}}_password">Password</label>
              <input type="password"
                class="form-control @error('password_confirmation') is-invalid @enderror" name="password_confirmation" id="{{$this->uniqid}}password_confirmation" aria-describedby="helpId" placeholder="confirm password" wire:model="password_confirmation" autocomplete="new-password">
                <small id="{{$this->uniqid}}_helpId" class="form-text text-muted">change your password</small>
                @error('password_confirmation') <div  class="invalid-feedback">{{ $message }}</div> @enderror
              <input type="password"
                class="form-control @error('password') is-invalid @enderror" name="password" id="{{$this->uniqid}}password" aria-describedby="helpId" placeholder="password" wire:model="password" autocomplete="new-password">
              @error('password') <div  class="invalid-feedback">{{ $message }}</div> @enderror
            </div>
            <button type="button" class="btn btn-primary" wire:click="addUser">AddUser</button>
        </form>
    </div>
</div>


```

**Comment by pedzed** [comment]
@zajinx Please format with ` ```blade` rather than just ` ``` `, so that syntax highlighting is added.

I can confirm that it is back. I'm not sure what causes it though. It's strange and a bit frustrating.

**Comment by pedzed** [comment]
OK, I'm going nuts. How do you even debug this thing?
```js
index.js:20 Uncaught (in promise) TypeError: Cannot read property 'data' of null
    at new Component (index.js:20)
    at onNodeAdded (index.js:334)
    at callHook (morphdom.js:35)
    at handleNodeAdded (morphdom.js:140)
    at handleNodeAdded (morphdom.js:164)
    at morphdom.js:407
    at morphEl (morphdom.js:219)
    at morphdom.js:463
    at Component.value (index.js:249)
    at Component.value (index.js:222)
    at Component.value (index.js:167)
    at Component.value (index.js:143)
    at Connection.value (index.js:30)
    at Object.Connection.driver.onMessage (index.js:9)
    at http.js:30
```

**Comment by uteqnl** [comment]
@pedzed In your template, did you add a :key? This solved my problem that threw the same error:

Parent Livewire component view
```blade
@foreach ($items as $key => $item)

    <livewire:child-component :key="$key" />

@endforeach
```

Before add the :key I got the same error. After that it was resolved. @calebporzio if turns out to be the issue. Would it be wise to more explicitly state on the [Nesting Components page](https://laravel-livewire.com/docs/nesting-components) that it is required to add :key of you add child components? Or maybe catch the error and throw a clear error?

I would be glad to create a pull-request if there are more people interested in this.
